Sources/Variable.swift

```diff
@@ -87,19 +105,9 @@ public struct Variable : Equatable, Resolvable {
           current = dictionary[bit]
         }
       } else if let array = current as? [Any] {
-        if let index = Int(bit) {
-          if index >= 0 && index < array.count {
-            current = array[index]
-          } else {
-            current = nil
-          }
-        } else if bit == "first" {
-          current = array.first
-        } else if bit == "last" {
-          current = array.last
-        } else if bit == "count" {
-          current = array.count
-        }
+        current = resolveCollection(array, bit: bit)
```

Comment:
Would it be worth changing the test to just `if let collection = current as? Collection {`, or would that be too broad of a cast?

Reply:
`Collection` is a generic protocol so can't be used like that.

Reply:
Right 🤦‍♂️